<!DOCTYPE html>
<html>
<head>
  <meta charset="utf-8">
  <meta name="generator" content="pandoc">
  <meta name="author" content="Neil D. Lawrence">
  <meta name="dcterms.date" content="2015-11-17">
  <title>Dimensionality Reduction: Latent Variable Modelling</title>
  <meta name="apple-mobile-web-app-capable" content="yes">
  <meta name="apple-mobile-web-app-status-bar-style" content="black-translucent">
  <meta name="viewport" content="width=device-width, initial-scale=1.0, maximum-scale=1.0, user-scalable=no, minimal-ui">
  <link rel="stylesheet" href="reveal.js/css/reveal.css">
  <style type="text/css">
      code{white-space: pre-wrap;}
      span.smallcaps{font-variant: small-caps;}
      span.underline{text-decoration: underline;}
      div.column{display: inline-block; vertical-align: top; width: 50%;}
  </style>
  <link rel="stylesheet" href="reveal.js/css/theme/black.css" id="theme">
  <link rel="stylesheet" href="talks.css"/>
  <!-- Printing and PDF exports -->
  <script>
    var link = document.createElement( 'link' );
    link.rel = 'stylesheet';
    link.type = 'text/css';
    link.href = window.location.search.match( /print-pdf/gi ) ? 'reveal.js/css/print/pdf.css' : 'reveal.js/css/print/paper.css';
    document.getElementsByTagName( 'head' )[0].appendChild( link );
  </script>
  <!--[if lt IE 9]>
  <script src="reveal.js/lib/js/html5shiv.js"></script>
  <![endif]-->
  <script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.1/MathJax.js?config=TeX-AMS-MML_SVG" type="text/javascript"></script>
  <script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.1/MathJax.js?config=TeX-AMS-MML_SVG" type="text/javascript"></script>
  <script type="text/x-mathjax-config">
    MathJax.Hub.Config({
    TeX: {
         extensions: ["color.js"]
      }
    });
  </script>
  <script src="figure-animate.js"></script>
</head>
<body>
\[\newcommand{\Amatrix}{\mathbf{A}}
\newcommand{\KL}[2]{\text{KL}\left( #1\,\|\,#2 \right)}
\newcommand{\Kaast}{\kernelMatrix_{\mathbf{ \ast}\mathbf{ \ast}}}
\newcommand{\Kastu}{\kernelMatrix_{\mathbf{ \ast} \inducingVector}}
\newcommand{\Kff}{\kernelMatrix_{\mappingFunctionVector \mappingFunctionVector}}
\newcommand{\Kfu}{\kernelMatrix_{\mappingFunctionVector \inducingVector}}
\newcommand{\Kuast}{\kernelMatrix_{\inducingVector \bf\ast}}
\newcommand{\Kuf}{\kernelMatrix_{\inducingVector \mappingFunctionVector}}
\newcommand{\Kuu}{\kernelMatrix_{\inducingVector \inducingVector}}
\newcommand{\Kuui}{\Kuu^{-1}}
\newcommand{\Qaast}{\mathbf{Q}_{\bf \ast \ast}}
\newcommand{\Qastf}{\mathbf{Q}_{\ast \mappingFunction}}
\newcommand{\Qfast}{\mathbf{Q}_{\mappingFunctionVector \bf \ast}}
\newcommand{\Qff}{\mathbf{Q}_{\mappingFunctionVector \mappingFunctionVector}}
\newcommand{\aMatrix}{\mathbf{A}}
\newcommand{\aScalar}{a}
\newcommand{\aVector}{\mathbf{a}}
\newcommand{\acceleration}{a}
\newcommand{\bMatrix}{\mathbf{B}}
\newcommand{\bScalar}{b}
\newcommand{\bVector}{\mathbf{b}}
\newcommand{\basisFunc}{\phi}
\newcommand{\basisFuncVector}{\boldsymbol{ \basisFunc}}
\newcommand{\basisFunction}{\phi}
\newcommand{\basisLocation}{\mu}
\newcommand{\basisMatrix}{\boldsymbol{ \Phi}}
\newcommand{\basisScalar}{\basisFunction}
\newcommand{\basisVector}{\boldsymbol{ \basisFunction}}
\newcommand{\activationFunction}{\phi}
\newcommand{\activationMatrix}{\boldsymbol{ \Phi}}
\newcommand{\activationScalar}{\basisFunction}
\newcommand{\activationVector}{\boldsymbol{ \basisFunction}}
\newcommand{\bigO}{\mathcal{O}}
\newcommand{\binomProb}{\pi}
\newcommand{\cMatrix}{\mathbf{C}}
\newcommand{\cbasisMatrix}{\hat{\boldsymbol{ \Phi}}}
\newcommand{\cdataMatrix}{\hat{\dataMatrix}}
\newcommand{\cdataScalar}{\hat{\dataScalar}}
\newcommand{\cdataVector}{\hat{\dataVector}}
\newcommand{\centeredKernelMatrix}{\mathbf{ \MakeUppercase{\centeredKernelScalar}}}
\newcommand{\centeredKernelScalar}{b}
\newcommand{\centeredKernelVector}{\centeredKernelScalar}
\newcommand{\centeringMatrix}{\mathbf{H}}
\newcommand{\chiSquaredDist}[2]{\chi_{#1}^{2}\left(#2\right)}
\newcommand{\chiSquaredSamp}[1]{\chi_{#1}^{2}}
\newcommand{\conditionalCovariance}{\boldsymbol{ \Sigma}}
\newcommand{\coregionalizationMatrix}{\mathbf{B}}
\newcommand{\coregionalizationScalar}{b}
\newcommand{\coregionalizationVector}{\mathbf{ \coregionalizationScalar}}
\newcommand{\covDist}[2]{\text{cov}_{#2}\left(#1\right)}
\newcommand{\covSamp}[1]{\text{cov}\left(#1\right)}
\newcommand{\covarianceScalar}{c}
\newcommand{\covarianceVector}{\mathbf{ \covarianceScalar}}
\newcommand{\covarianceMatrix}{\mathbf{C}}
\newcommand{\covarianceMatrixTwo}{\boldsymbol{ \Sigma}}
\newcommand{\croupierScalar}{s}
\newcommand{\croupierVector}{\mathbf{ \croupierScalar}}
\newcommand{\croupierMatrix}{\mathbf{ \MakeUppercase{\croupierScalar}}}
\newcommand{\dataDim}{p}
\newcommand{\dataIndex}{i}
\newcommand{\dataIndexTwo}{j}
\newcommand{\dataMatrix}{\mathbf{Y}}
\newcommand{\dataScalar}{y}
\newcommand{\dataSet}{\mathcal{D}}
\newcommand{\dataStd}{\sigma}
\newcommand{\dataVector}{\mathbf{ \dataScalar}}
\newcommand{\decayRate}{d}
\newcommand{\degreeMatrix}{\mathbf{ \MakeUppercase{\degreeScalar}}}
\newcommand{\degreeScalar}{d}
\newcommand{\degreeVector}{\mathbf{ \degreeScalar}}
% Already defined by latex
%\newcommand{\det}[1]{\left|#1\right|}
\newcommand{\diag}[1]{\text{diag}\left(#1\right)}
\newcommand{\diagonalMatrix}{\mathbf{D}}
\newcommand{\diff}[2]{\frac{\text{d}#1}{\text{d}#2}}
\newcommand{\diffTwo}[2]{\frac{\text{d}^2#1}{\text{d}#2^2}}
\newcommand{\displacement}{x}
\newcommand{\displacementVector}{\textbf{\displacement}}
\newcommand{\distanceMatrix}{\mathbf{ \MakeUppercase{\distanceScalar}}}
\newcommand{\distanceScalar}{d}
\newcommand{\distanceVector}{\mathbf{ \distanceScalar}}
\newcommand{\eigenvaltwo}{\ell}
\newcommand{\eigenvaltwoMatrix}{\mathbf{L}}
\newcommand{\eigenvaltwoVector}{\mathbf{l}}
\newcommand{\eigenvalue}{\lambda}
\newcommand{\eigenvalueMatrix}{\boldsymbol{ \Lambda}}
\newcommand{\eigenvalueVector}{\boldsymbol{ \lambda}}
\newcommand{\eigenvector}{\mathbf{ \eigenvectorScalar}}
\newcommand{\eigenvectorMatrix}{\mathbf{U}}
\newcommand{\eigenvectorScalar}{u}
\newcommand{\eigenvectwo}{\mathbf{v}}
\newcommand{\eigenvectwoMatrix}{\mathbf{V}}
\newcommand{\eigenvectwoScalar}{v}
\newcommand{\entropy}[1]{\mathcal{H}\left(#1\right)}
\newcommand{\errorFunction}{E}
\newcommand{\expDist}[2]{\left<#1\right>_{#2}}
\newcommand{\expSamp}[1]{\left<#1\right>}
\newcommand{\expectation}[1]{\left\langle #1 \right\rangle }
\newcommand{\expectationDist}[2]{\left\langle #1 \right\rangle _{#2}}
\newcommand{\expectedDistanceMatrix}{\mathcal{D}}
\newcommand{\eye}{\mathbf{I}}
\newcommand{\fantasyDim}{r}
\newcommand{\fantasyMatrix}{\mathbf{ \MakeUppercase{\fantasyScalar}}}
\newcommand{\fantasyScalar}{z}
\newcommand{\fantasyVector}{\mathbf{ \fantasyScalar}}
\newcommand{\featureStd}{\varsigma}
\newcommand{\gammaCdf}[3]{\mathcal{GAMMA CDF}\left(#1|#2,#3\right)}
\newcommand{\gammaDist}[3]{\mathcal{G}\left(#1|#2,#3\right)}
\newcommand{\gammaSamp}[2]{\mathcal{G}\left(#1,#2\right)}
\newcommand{\gaussianDist}[3]{\mathcal{N}\left(#1|#2,#3\right)}
\newcommand{\gaussianSamp}[2]{\mathcal{N}\left(#1,#2\right)}
\newcommand{\given}{|}
\newcommand{\half}{\frac{1}{2}}
\newcommand{\heaviside}{H}
\newcommand{\hiddenMatrix}{\mathbf{ \MakeUppercase{\hiddenScalar}}}
\newcommand{\hiddenScalar}{h}
\newcommand{\hiddenVector}{\mathbf{ \hiddenScalar}}
\newcommand{\identityMatrix}{\eye}
\newcommand{\inducingInputScalar}{z}
\newcommand{\inducingInputVector}{\mathbf{ \inducingInputScalar}}
\newcommand{\inducingInputMatrix}{\mathbf{Z}}
\newcommand{\inducingScalar}{u}
\newcommand{\inducingVector}{\mathbf{ \inducingScalar}}
\newcommand{\inducingMatrix}{\mathbf{U}}
\newcommand{\inlineDiff}[2]{\text{d}#1/\text{d}#2}
\newcommand{\inputDim}{q}
\newcommand{\inputMatrix}{\mathbf{X}}
\newcommand{\inputScalar}{x}
\newcommand{\inputSpace}{\mathcal{X}}
\newcommand{\inputVals}{\inputVector}
\newcommand{\inputVector}{\mathbf{ \inputScalar}}
\newcommand{\iterNum}{k}
\newcommand{\kernel}{\kernelScalar}
\newcommand{\kernelMatrix}{\mathbf{K}}
\newcommand{\kernelScalar}{k}
\newcommand{\kernelVector}{\mathbf{ \kernelScalar}}
\newcommand{\kff}{\kernelScalar_{\mappingFunction \mappingFunction}}
\newcommand{\kfu}{\kernelVector_{\mappingFunction \inducingScalar}}
\newcommand{\kuf}{\kernelVector_{\inducingScalar \mappingFunction}}
\newcommand{\kuu}{\kernelVector_{\inducingScalar \inducingScalar}}
\newcommand{\lagrangeMultiplier}{\lambda}
\newcommand{\lagrangeMultiplierMatrix}{\boldsymbol{ \Lambda}}
\newcommand{\lagrangian}{L}
\newcommand{\laplacianFactor}{\mathbf{ \MakeUppercase{\laplacianFactorScalar}}}
\newcommand{\laplacianFactorScalar}{m}
\newcommand{\laplacianFactorVector}{\mathbf{ \laplacianFactorScalar}}
\newcommand{\laplacianMatrix}{\mathbf{L}}
\newcommand{\laplacianScalar}{\ell}
\newcommand{\laplacianVector}{\mathbf{ \ell}}
\newcommand{\latentDim}{q}
\newcommand{\latentDistanceMatrix}{\boldsymbol{ \Delta}}
\newcommand{\latentDistanceScalar}{\delta}
\newcommand{\latentDistanceVector}{\boldsymbol{ \delta}}
\newcommand{\latentForce}{f}
\newcommand{\latentFunction}{u}
\newcommand{\latentFunctionVector}{\mathbf{ \latentFunction}}
\newcommand{\latentFunctionMatrix}{\mathbf{ \MakeUppercase{\latentFunction}}}
\newcommand{\latentIndex}{j}
\newcommand{\latentScalar}{z}
\newcommand{\latentVector}{\mathbf{ \latentScalar}}
\newcommand{\latentMatrix}{\mathbf{Z}}
\newcommand{\learnRate}{\eta}
\newcommand{\lengthScale}{\ell}
\newcommand{\rbfWidth}{\ell}
\newcommand{\likelihoodBound}{\mathcal{L}}
\newcommand{\likelihoodFunction}{L}
\newcommand{\locationScalar}{\mu}
\newcommand{\locationVector}{\boldsymbol{ \locationScalar}}
\newcommand{\locationMatrix}{\mathbf{M}}
\newcommand{\variance}[1]{\text{var}\left( #1 \right)}
\newcommand{\mappingFunction}{f}
\newcommand{\mappingFunctionMatrix}{\mathbf{F}}
\newcommand{\mappingFunctionTwo}{g}
\newcommand{\mappingFunctionTwoMatrix}{\mathbf{G}}
\newcommand{\mappingFunctionTwoVector}{\mathbf{ \mappingFunctionTwo}}
\newcommand{\mappingFunctionVector}{\mathbf{ \mappingFunction}}
\newcommand{\scaleScalar}{s}
\newcommand{\mappingScalar}{w}
\newcommand{\mappingVector}{\mathbf{ \mappingScalar}}
\newcommand{\mappingMatrix}{\mathbf{W}}
\newcommand{\mappingScalarTwo}{v}
\newcommand{\mappingVectorTwo}{\mathbf{ \mappingScalarTwo}}
\newcommand{\mappingMatrixTwo}{\mathbf{V}}
\newcommand{\maxIters}{K}
\newcommand{\meanMatrix}{\mathbf{M}}
\newcommand{\meanScalar}{\mu}
\newcommand{\meanTwoMatrix}{\mathbf{M}}
\newcommand{\meanTwoScalar}{m}
\newcommand{\meanTwoVector}{\mathbf{ \meanTwoScalar}}
\newcommand{\meanVector}{\boldsymbol{ \meanScalar}}
\newcommand{\mrnaConcentration}{m}
\newcommand{\naturalFrequency}{\omega}
\newcommand{\neighborhood}[1]{\mathcal{N}\left( #1 \right)}
\newcommand{\neilurl}{http://inverseprobability.com/}
\newcommand{\noiseMatrix}{\boldsymbol{ E}}
\newcommand{\noiseScalar}{\epsilon}
\newcommand{\noiseVector}{\boldsymbol{ \epsilon}}
\newcommand{\norm}[1]{\left\Vert #1 \right\Vert}
\newcommand{\normalizedLaplacianMatrix}{\hat{\mathbf{L}}}
\newcommand{\normalizedLaplacianScalar}{\hat{\ell}}
\newcommand{\normalizedLaplacianVector}{\hat{\mathbf{ \ell}}}
\newcommand{\numActive}{m}
\newcommand{\numBasisFunc}{m}
\newcommand{\numComponents}{m}
\newcommand{\numComps}{K}
\newcommand{\numData}{n}
\newcommand{\numFeatures}{K}
\newcommand{\numHidden}{h}
\newcommand{\numInducing}{m}
\newcommand{\numLayers}{\ell}
\newcommand{\numNeighbors}{K}
\newcommand{\numSequences}{s}
\newcommand{\numSuccess}{s}
\newcommand{\numTasks}{m}
\newcommand{\numTime}{T}
\newcommand{\numTrials}{S}
\newcommand{\outputIndex}{j}
\newcommand{\paramVector}{\boldsymbol{ \theta}}
\newcommand{\parameterMatrix}{\boldsymbol{ \Theta}}
\newcommand{\parameterScalar}{\theta}
\newcommand{\parameterVector}{\boldsymbol{ \parameterScalar}}
\newcommand{\partDiff}[2]{\frac{\partial#1}{\partial#2}}
\newcommand{\precisionScalar}{j}
\newcommand{\precisionVector}{\mathbf{ \precisionScalar}}
\newcommand{\precisionMatrix}{\mathbf{J}}
\newcommand{\pseudotargetScalar}{\widetilde{y}}
\newcommand{\pseudotargetVector}{\mathbf{ \pseudotargetScalar}}
\newcommand{\pseudotargetMatrix}{\mathbf{ \widetilde{Y}}}
\newcommand{\rank}[1]{\text{rank}\left(#1\right)}
\newcommand{\rayleighDist}[2]{\mathcal{R}\left(#1|#2\right)}
\newcommand{\rayleighSamp}[1]{\mathcal{R}\left(#1\right)}
\newcommand{\responsibility}{r}
\newcommand{\rotationScalar}{r}
\newcommand{\rotationVector}{\mathbf{ \rotationScalar}}
\newcommand{\rotationMatrix}{\mathbf{R}}
\newcommand{\sampleCovScalar}{s}
\newcommand{\sampleCovVector}{\mathbf{ \sampleCovScalar}}
\newcommand{\sampleCovMatrix}{\mathbf{s}}
\newcommand{\scalarProduct}[2]{\left\langle{#1},{#2}\right\rangle}
\newcommand{\sign}[1]{\text{sign}\left(#1\right)}
\newcommand{\sigmoid}[1]{\sigma\left(#1\right)}
\newcommand{\singularvalue}{\ell}
\newcommand{\singularvalueMatrix}{\mathbf{L}}
\newcommand{\singularvalueVector}{\mathbf{l}}
\newcommand{\sorth}{\mathbf{u}}
\newcommand{\spar}{\lambda}
\newcommand{\trace}[1]{\text{tr}\left(#1\right)}
\newcommand{\BasalRate}{B}
\newcommand{\DampingCoefficient}{C}
\newcommand{\DecayRate}{D}
\newcommand{\Displacement}{X}
\newcommand{\LatentForce}{F}
\newcommand{\Mass}{M}
\newcommand{\Sensitivity}{S}
\newcommand{\basalRate}{b}
\newcommand{\dampingCoefficient}{c}
\newcommand{\mass}{m}
\newcommand{\sensitivity}{s}
\newcommand{\springScalar}{\kappa}
\newcommand{\springVector}{\boldsymbol{ \kappa}}
\newcommand{\springMatrix}{\boldsymbol{ \mathcal{K}}}
\newcommand{\tfConcentration}{p}
\newcommand{\tfDecayRate}{\delta}
\newcommand{\tfMrnaConcentration}{f}
\newcommand{\tfVector}{\mathbf{ \tfConcentration}}
\newcommand{\velocity}{v}
\newcommand{\sufficientStatsScalar}{g}
\newcommand{\sufficientStatsVector}{\mathbf{ \sufficientStatsScalar}}
\newcommand{\sufficientStatsMatrix}{\mathbf{G}}
\newcommand{\switchScalar}{s}
\newcommand{\switchVector}{\mathbf{ \switchScalar}}
\newcommand{\switchMatrix}{\mathbf{S}}
\newcommand{\tr}[1]{\text{tr}\left(#1\right)}
\newcommand{\loneNorm}[1]{\left\Vert #1 \right\Vert_1}
\newcommand{\ltwoNorm}[1]{\left\Vert #1 \right\Vert_2}
\newcommand{\onenorm}[1]{\left\vert#1\right\vert_1}
\newcommand{\twonorm}[1]{\left\Vert #1 \right\Vert}
\newcommand{\vScalar}{v}
\newcommand{\vVector}{\mathbf{v}}
\newcommand{\vMatrix}{\mathbf{V}}
\newcommand{\varianceDist}[2]{\text{var}_{#2}\left( #1 \right)}
% Already defined by latex
%\newcommand{\vec}{#1:}
\newcommand{\vecb}[1]{\left(#1\right):}
\newcommand{\weightScalar}{w}
\newcommand{\weightVector}{\mathbf{ \weightScalar}}
\newcommand{\weightMatrix}{\mathbf{W}}
\newcommand{\weightedAdjacencyMatrix}{\mathbf{A}}
\newcommand{\weightedAdjacencyScalar}{a}
\newcommand{\weightedAdjacencyVector}{\mathbf{ \weightedAdjacencyScalar}}
\newcommand{\onesVector}{\mathbf{1}}
\newcommand{\zerosVector}{\mathbf{0}}
\]
  <div class="reveal">
    <div class="slides">

<section id="title-slide">
  <h1 class="title">Dimensionality Reduction: Latent Variable Modelling</h1>
  <p class="author" style="text-align:center"> </p>
  <p class="date" style="text-align:center"><time>2015-11-17</time></p>
  <p class="venue" style="text-align:center">University of Sheffield</p>
</section>

<section class="slide level3">

<!-- Front matter -->
<!---->
<!--Back matter-->
<p>.</p>
<!-- The last names to be defined. Should be defined entirely in terms of macros from above-->
</section>
<section id="review" class="slide level3">
<h3>Review</h3>
<ul>
<li>Last time: Looked at Bayesian Regression.</li>
<li>Introduced priors and marginal likelihoods.</li>
<li>This time: Unsupervised Learning</li>
</ul>
</section>
<section id="unsupervised-learning" class="slide level3">
<h3>Unsupervised Learning</h3>
<ul>
<li>Supervised learning is learning where each data has a label (e.g. regression output)</li>
<li>In unsupervised learning we have no labels for the data.</li>
<li>Often thought of as structure discovery.</li>
<li>Finding features in the data</li>
<li>Exploratory data analysis</li>
</ul>
</section>
<section id="clustering" class="slide level3">
<h3>Clustering</h3>
<ul>
<li>One common approach, not deeply covered in this course.</li>
<li>Associate each data point, <span class="math inline">\(\dataVector_{i, :}\)</span> with one of <span class="math inline">\(k\)</span> different discrete groups.</li>
<li>For example:</li>
<li>Clustering animals into discrete groups. Are animals discrete or continuous?</li>
<li>Clustering into different different <em>political</em> affiliations.</li>
<li>Humans do seem to like clusters:</li>
<li>Very useful when interacting with biologists.</li>
<li>Subtle difference between clustering and <em>vector quantisation</em></li>
</ul>
</section>
<section id="trying-to-teach-about-infinity" class="slide level3">
<h3>Trying to Teach About Infinity</h3>
<ul>
<li>Little anecdote.</li>
</ul>
</section>
<section id="clustering-and-vector-quantisation" class="slide level3">
<h3>Clustering and Vector Quantisation</h3>
<ul>
<li>To my mind difference is in clustering there should be a reduction in data density between samples.</li>
<li>This definition is not universally applied.</li>
<li>For today’s purposes we merge them:</li>
<li>Determine how to allocate each point to a group and <em>harder</em> total number of groups.</li>
</ul>
</section>
<section id="k-means-clustering" class="slide level3">
<h3><span class="math inline">\(k\)</span>-means Clustering</h3>
<ul>
<li>Simple algorithm for allocating points to groups.</li>
<li><em>Require</em>: Set of <span class="math inline">\(k\)</span> cluster centres &amp; assignment of each points to a cluster.</li>
</ul>
<ol type="1">
<li>Initialize cluster centres as randomly selected data points.
<ol start="2" type="1">
<li>Assign each data point to <em>nearest</em> cluster centre.</li>
<li>Update each cluster centre by setting it to the mean of assigned data points.</li>
<li>Repeat 2 and 3 until cluster allocations do not change.</li>
</ol></li>
</ol>
</section>
<section id="objective-function" class="slide level3">
<h3>Objective Function</h3>
<ul>
<li>This minimizes the objective <span class="math display">\[
  E=\sum_{j=1}^K \sum_{i\ \text{allocated to}\ j}  \left(\dataVector_{i, :} - \boldsymbol{\mu}_{j, :}\right)^\top\left(\dataVector_{i, :} - \boldsymbol{\mu}_{j, :}\right)
  \]</span> <em>i.e.</em> it minimizes thesum of Euclidean squared distances betwen points and their associated centres.</li>
<li>The minimum is <em>not</em> guaranteed to be <em>global</em> or <em>unique</em>.</li>
<li>This objective is a non-convex optimization problem.</li>
</ul>
</section>
<section id="other-clustering-approaches" class="slide level3">
<h3>Other Clustering Approaches</h3>
<ul>
<li>Spectral clustering (<span class="citation" data-cites="Shi:normalized00">Shi and Malik (2000)</span>,<span class="citation" data-cites="Ng:spectral02">Ng et al. (n.d.)</span>)</li>
<li>Allows clusters which aren’t convex hulls.</li>
<li>Dirichlet process</li>
<li>A probabilistic formulation for a clustering algorithm that is <em>non-parametric</em>.</li>
<li>Loosely speaking it allows infinite clusters</li>
<li>In practice useful for dealing with previously unknown species (e.g. a “Black Swan Event”).</li>
</ul>
</section>
<section id="high-dimensional-data" class="slide level3">
<h3>High Dimensional Data</h3>
<ul>
<li>USPS Data Set Handwritten Digit</li>
<li>3648 dimensions (64 rows, 57 columns)</li>
<li>Space contains much more than just this digit.</li>
</ul>
</section>
<section id="usps-samples" class="slide level3">
<h3>USPS Samples</h3>
<ul>
<li>Even if we sample every nanonsecond from now until end of universe you won’t see original six!</li>
</ul>
</section>
<section id="simple-model-of-digit" class="slide level3">
<h3>Simple Model of Digit</h3>
<ul>
<li>Rotate a prototype</li>
</ul>
</section>
<section id="low-dimensional-manifolds" class="slide level3">
<h3>Low Dimensional Manifolds</h3>
<ul>
<li>Pure rotation is too simple</li>
<li>In practice data may undergo several distortions.</li>
<li>For high dimensional data with <em>structure</em>:</li>
<li>We expect fewer distortions than dimensions;</li>
<li>Therefore we expect the data to live on a lower dimensional manifold.</li>
<li>Conclusion: Deal with high dimensional data by looking for a lower dimensional non-linear embedding.</li>
</ul>
</section>
<section id="principal-component-analysis" class="slide level3">
<h3>Principal Component Analysis</h3>
<ul>
<li>PCA (<span class="citation" data-cites="Hotelling:analysis33">Hotelling (1933)</span>) is a linear embedding.</li>
<li>Today its presented as:</li>
<li>Rotate to find ‘directions’ in data with maximal variance.</li>
<li>How do we find these directions?</li>
<li>Algorithmically we do this by diagonalizing the sample covariance matrix <span class="math display">\[
  \mathbf{S}=\frac{1}{\numData}\sum_{i=1}^n \left(\dataVector_{i, :}-\boldsymbol{\mu}\right)\left(\dataVector_{i, :} - \boldsymbol{\mu}\right)^\top
  \]</span></li>
</ul>
</section>
<section id="principal-component-analysis-1" class="slide level3">
<h3>Principal Component Analysis</h3>
<ul>
<li>Find directions in the data, <span class="math inline">\(\mathbf{x} = \mathbf{U}\dataVector\)</span>, for which variance is maximized.</li>
</ul>
</section>
<section id="lagrangian" class="slide level3">
<h3>Lagrangian</h3>
<ul>
<li><p>Solution is found via constrained optimisation (which uses <a href="https://en.wikipedia.org/wiki/Lagrange_multiplier">Lagrange multipliers</a>): <span class="math display">\[
  L\left(\mathbf{u}_{1},\lambda_{1}\right)=\mathbf{u}_{1}^{\top}\mathbf{S}\mathbf{u}_{1}+\lambda_{1}\left(1-\mathbf{u}_{1}^{\top}\mathbf{u}_{1}\right)
  \]</span></p></li>
<li>Gradient with respect to <span class="math inline">\(\mathbf{u}_{1}\)</span> <span class="math display">\[\frac{\text{d}L\left(\mathbf{u}_{1},\lambda_{1}\right)}{\text{d}\mathbf{u}_{1}}=2\mathbf{S}\mathbf{u}_{1}-2\lambda_{1}\mathbf{u}_{1}\]</span> rearrange to form <span class="math display">\[\mathbf{S}\mathbf{u}_{1}=\lambda_{1}\mathbf{u}_{1}.\]</span> Which is known as an <a href="https://en.wikipedia.org/wiki/Eigenvalues_and_eigenvectors"><em>eigenvalue problem</em></a>.</li>
<li><p>Further directions that are <em>orthogonal</em> to the first can also be shown to be eigenvectors of the covariance.</p></li>
</ul>
</section>
<section id="linear-dimensionality-reduction" class="slide level3">
<h3>Linear Dimensionality Reduction</h3>
<ul>
<li>Represent data, <span class="math inline">\(\dataMatrix\)</span>, with a lower dimensional set of latent variables <span class="math inline">\(\latentMatrix\)</span>.</li>
<li>Assume a linear relationship of the form <span class="math display">\[
  \dataVector_{i,:}=\mappingMatrix\mathbf{x}_{i,:}+\boldsymbol{\epsilon}_{i,:},
  \]</span> where <span class="math display">\[
  \boldsymbol{\epsilon}_{i,:} \sim \mathcal{N}(\mathbf{0}, \sigma^2\mathbf{I})
  \]</span></li>
</ul>
</section>
<section id="linear-latent-variable-model" class="slide level3">
<h3>Linear Latent Variable Model</h3>
<p><strong>Probabilistic PCA</strong></p>
<ul>
<li>Define <em>linear-Gaussian relationship</em> between latent variables and data.</li>
<li><strong>Standard</strong> Latent variable approach:</li>
<li>Define Gaussian prior over <em>latent space</em>, <span class="math inline">\(\latentMatrix\)</span>.</li>
<li>Integrate out <em>latent variables</em>.</li>
</ul>

<p><span class="math display">\[p\left(\dataMatrix|\latentMatrix,\mappingMatrix\right)=\prod_{i=1}^{\numData}\mathcal{N}\left(\dataVector_{i,:}|\mappingMatrix\mathbf{x}_{i,:},
\sigma^2\mathbf{I}\right)\]</span></p>
<p><span class="math display">\[p\left(\latentMatrix\right)=\prod_{i=1}^{\numData}\mathcal{N}\left(\mathbf{x}_{i,:}|\mathbf{0},\mathbf{I}\right)\]</span></p>
<p><span class="math display">\[p\left(\dataMatrix|\mappingMatrix\right)=\prod_{i=1}^{\numData}\mathcal{N}\left(\dataVector_{i,:}|\mathbf{0},\mappingMatrix\mappingMatrix^{\top}+\sigma^{2}\mathbf{I}\right)\]</span></p>
</section>
<section id="computation-of-the-marginal-likelihood" class="slide level3">
<h3>Computation of the Marginal Likelihood</h3>
<p><span class="math display">\[\dataVector_{i,:}=\mappingMatrix\mathbf{x}_{i,:}+\boldsymbol{\epsilon}_{i,:},\quad \mathbf{x}_{i,:} \sim \mathcal{N}(\mathbf{0},\mathbf{I}), \quad \boldsymbol{\epsilon}_{i,:} \sim \mathcal{N}(\mathbf{0},\sigma^{2}\mathbf{I})\]</span></p>
<p><span class="math display">\[\mappingMatrix\mathbf{x}_{i,:} \sim \mathcal{N}(\mathbf{0},\mappingMatrix\mappingMatrix^\top)\]</span></p>
<p><span class="math display">\[\mappingMatrix\mathbf{x}_{i, :} + \boldsymbol{\epsilon}_{i, :} \sim \mathcal{N}\left(\mathbf{0},\mappingMatrix\mappingMatrix^\top + \sigma^2 \mathbf{I}\right)\]</span></p>
</section>
<section id="linear-latent-variable-model-ii" class="slide level3">
<h3>Linear Latent Variable Model II</h3>
<p><strong>Probabilistic PCA Max. Likelihood Soln</strong> (<span class="citation" data-cites="Tipping:probpca99">Tipping and Bishop (1999)</span>)</p>
%

<p><span class="math display">\[p\left(\dataMatrix|\mappingMatrix\right)=\prod_{i=1}^{\numData}\mathcal{N}\left(\dataVector_{i, :}|\mathbf{0}, \mappingMatrix\mappingMatrix^{\top}+\sigma^{2}\mathbf{I}\right)\]</span></p>
</section>
<section id="linear-latent-variable-model-ii-1" class="slide level3">
<h3>Linear Latent Variable Model II</h3>
<p><strong>Probabilistic PCA Max. Likelihood Soln</strong> (<span class="citation" data-cites="Tipping:probpca99">Tipping and Bishop (1999)</span>) <span class="math display">\[
  p\left(\dataMatrix|\mappingMatrix\right)=\prod_{i=1}^{\numData}\mathcal{N}\left(\dataVector_{i,:}|\mathbf{0},\mathbf{C}\right),\quad \mathbf{C}=\mappingMatrix\mappingMatrix^{\top}+\sigma^{2}\mathbf{I}
  \]</span> <span class="math display">\[
  \log p\left(\dataMatrix|\mappingMatrix\right)=-\frac{\numData}{2}\log\left|\mathbf{C}\right|-\frac{1}{2}\text{tr}\left(\mathbf{C}^{-1}\dataMatrix^{\top}\dataMatrix\right)+\text{const.}
  \]</span> If <span class="math inline">\(\mathbf{U}_{q}\)</span> are first <span class="math inline">\(q\)</span> principal eigenvectors of <span class="math inline">\(n^{-1}\dataMatrix^{\top}\dataMatrix\)</span> and the corresponding eigenvalues are <span class="math inline">\(\boldsymbol{\Lambda}_{q}\)</span>, <span class="math display">\[
  \mappingMatrix=\mathbf{U}_{q}\mathbf{L}\mathbf{R}^{\top},\quad\mathbf{L}=\left(\boldsymbol{\Lambda}_{q}-\sigma^{2}\mathbf{I}\right)^{\frac{1}{2}}
  \]</span> where <span class="math inline">\(\mathbf{R}\)</span> is an arbitrary rotation matrix.</p>
</section>
<section id="reading" class="slide level3">
<h3>Reading</h3>
<ul>
<li>Chapter 7 of <span class="citation" data-cites="Rogers:book11">Rogers and Girolami (2011)</span> up to pg 249.</li>
</ul>
<p>So far in our classes we have focussed mainly on regression problems, which are examples of supervised learning. We have considered the relationship between the likelihood and the objective function and we have shown how we can find paramters by maximizing the likelihood (equivalent to minimizing the objective function) and in the last session we saw how we can <em>marginalize</em> the parameters in a process known as Bayesian inference.</p>
<p>Now we are going to turn to a different form of learning, commonly known as <em>unsupervised</em> learning. In unsupervised learning our data isn’t necessarily labelled in any form, but we want models that give us a better understanding of the data. We’ve actually seen an example of this already with <a href="./week2.ipynb"><em>matrix factorization</em> for collaborative filtering</a>, which we introduces in the context of <em>objective functions</em>. Now we will introduce a more probabilistic approach to such models, specifically we are interested in <em>latent variable</em> modelling.</p>
<!-- SECTION Latent Variables -->
<p>Latent means hidden, and hidden variables are simply <em>unobservable</em> variables. The idea of a latent variable is crucial to the concept of artificial intelligence, machine learning and experimental design. A latent variable could take many forms. We might observe a man walking along a road with a large bag of clothes and we might <em>infer</em> that the man is walking to the laundrette. Our observations are a highly complex data space, the response in our eyes is processed through our visual cortex, the combination of the indidivuals limb movememnts and the direction they are walking in all conflate in our heads to cause us to infer that (perhaps) the individual is going to the laundrette. We don’t <em>know</em> that the man is walking to the laundrette, but we have a model of the world that suggests that it’s a likely outcome for the very complex data. In some ways the latent variable can be seen as a <em>compression</em> of this very complex scene. If I were writing a book, I might write that “A man tripped over whilst walking to the laundrette”. In the reader’s mind an image of a man, perhaps laden with dirty clothes, may occur. All these ideas come from our expectations of the world around us. We can make further inference about the man, some of it perhaps plausible others less so. The man may be going to the laundrette because his washing machine is broken, or because he doesn’t have a large enough flat to have a washing machine, or because he’s carrying a duvet, or because he doesn’t like ironing. All of these may <em>increase</em> in probability given our observation, but they are still <em>latent</em> variables. Unless we follow the man back to his appartment, or start making other enquirires about the man, we don’t know the true answer.</p>
<p>It’s clear that to do inference about any complex system we <em>must</em> include latent variables. Latent variables are extremely powerful. In robotics, they are used to represent the <em>state</em> of the robot. The state of the robot may include its position (in x, y coordinates) its speed, its direction of facing. How are <em>these</em> variables unknown to the robot? Well the robot only posesses <em>sensors</em>, it can make observations of the nearest object in a certain direction, and it may have a map of its environment. If we represent the state of the robot as its position on a map, it may be uncertain of that position. If you go walking or running in the hills around Sheffield, you can take a very high quality ordnance survey map with you. However, unless you are a really excellent orienteer, when you are far from any given landmark, you will probably be <em>uncertain</em> about your true position on the map. These states are also latent variables.</p>
<p>In statistical analysis of experiments you try to control for each aspect of the experiment, in particular by <em>randomization</em>. So if I’m interested in the ability of a particular fertilizer to improve the yield of a particular plant I may design an experiment where I apply the fertilizer to some plants (the treatment group) and withold the fertilizer from others (the control group). I then test to see whether the yield from the treatment group is better (or worse) than the control group. I may find that I have an excellent yield for the treatment group. However, what if I’d (unknowlingly) planted all my treatment plants in a sunny part of the field, and all the control plants in a shady part of the field. That would also be a latent variable, in this case known as a <em>confounder</em>. In statistical experimental design <em>randomization</em> is used to attempt to eliminate the correlated effects of these confounders: you aim to ensure that if these confounders <em>do</em> exist their effects are not correlated with treatment and contorl. This is known as a <a href="http://en.wikipedia.org/wiki/Randomized_controlled_trial">randomized control trial</a>.</p>
<p>Greek philosophers worried a great deal about what was knowable and what was unknowable. Adherents of <a href="http://en.wikipedia.org/wiki/Skepticism">philosophical Skeptisism</a> were inspired by the idea that since your senses sometimes give you contradictory information, they cannot be trusted, and in extreme cases they chose to <em>ignore</em> their senses. This is an acknowledgement that very often the true state of the world cannot be known with precision. Unfortunately, these philosophers didn’t have a good understanding of probability, so they were unable to encapsulate their ideas through a <em>degree</em> of belief.</p>
<p>We often use language to express the compression of a complex behavior or patterns in a simpler way, for example we talk about motives as a useful distallation for a perhaps very complex patter of behavior. In physics we use principles of causation and simple laws to describe the world around us. Such motives or underlying principles are difficult to observe directly, our conclusions about them emerge over a period of time by observing indirect consequences of the latent variables.</p>
<p>Epistemic uncertainty allows us to deal with these worries by associating our degree of belief about the state of the world with a probaiblity distribution. This core idea underpins state space modelling, probabilistic graphical models and the wider field of latent variable modelling. In this session we are going to explore the idea in a simple linear system and see how it relates to <em>factor analysis</em> and <em>principal component analysis</em>.</p>
<!-- SECTION Your Personality -->
<p>At the beginning of the 20th century there was a great deal of interest amoungst psychologists in formalizing patterns of thought. The approach they used became known as factor analysis. The principle is that we observe a potentially high dimensional vector of characteristics about an individual. To formalize this, social scientists designed questionaires. We can envisage many questions that we may ask, but the assumption is that underlying these questions there are only a few traits that dictate the behavior. These models are known as latent trait models and the analysis is sometimes known as factor analysis. The idea is that there are a few characteristic traits that we are looking to discern. These traits or factors can be extracted by assimilating the high dimensional characteristics of the individual into a few latent factors.</p>
</section>
<section id="factor-analysis-model" class="slide level3">
<h3>Factor Analysis Model</h3>
<p>This causes us to consider a model as follows, if we are given a high dimensional vector of features (perhaps questionaire answers) associated with an individual, <span class="math inline">\(\dataVector\)</span>, we assume that these factors are actually generated from a low dimensional vector latent traits, or latent variables, which determine the personality. <span class="math display">\[
\dataVector = \mathbf{f}(\mathbf{x}) + \boldsymbol{\epsilon}
\]</span> where <span class="math inline">\(\mathbf{f}(\mathbf{x})\)</span> is a <em>vector valued</em> function that is dependent on the latent traits and <span class="math inline">\(\boldsymbol{\epsilon}\)</span> is some corrupting noise. For simplicity, we assume that the function is given by a <em>linear</em> relationship, <span class="math display">\[
\mathbf{f}(\mathbf{x}) = \mappingMatrix\mathbf{x}
\]</span> where we have introduced a matrix <span class="math inline">\(\mappingMatrix\)</span> that is sometimes referred to as the <em>factor loadings</em> but we also immediately see is related to our <em>multivariate linear regression</em> models from the <a href="./week3.ipynb">previous session on linear regression</a>. That is because our vector valued function is of the form <span class="math display">\[
\mathbf{f}(\mathbf{x}) =
\begin{bmatrix} f_1(\mathbf{x}) \\ f_2(\mathbf{x}) \\ \vdots \\
f_p(\mathbf{x})\end{bmatrix}
\]</span> where there are <span class="math inline">\(p\)</span> features associated with the individual. If we consider any of these functions individually we have a prediction function that looks like a regression model, <span class="math display">\[
f_j(\mathbf{x}) =
\mathbf{w}_{j, :}^\top \mathbf{x},
\]</span> for each element of the vector valued function, where <span class="math inline">\(\mathbf{w}_{:, j}\)</span> is the <span class="math inline">\(j\)</span>th column of the matrix <span class="math inline">\(\mappingMatrix\)</span>. In that context each column of <span class="math inline">\(\mappingMatrix\)</span> is a vector of <em>regression weights</em>. This is a multiple input and multiple output regression. Our inputs (or covariates) have dimensionality greater than 1 and our outputs (or response variables) also have dimensionality greater than one. Just as in a standard regression, we are assuming that we don’t observe the function directly (note that this <em>also</em> makes the function a <em>type</em> of latent variable), but we observe some corrupted variant of the function, where the corruption is given by <span class="math inline">\(\boldsymbol{\epsilon}\)</span>. Just as in linear regression we can assume that this corruption is given by Gaussian noise, where the noise for the <span class="math inline">\(j\)</span>th element of <span class="math inline">\(\dataVector\)</span> is by, <span class="math display">\[
\epsilon_j \sim \mathcal{N}(0, \sigma^2_j).
\]</span> Of course, just as in a regression problem we also need to make an assumption across the individual data points to form our full likelihood. Our data set now consists of many observations of <span class="math inline">\(\dataVector\)</span> for diffetent individuals. We store these observations in a <em>design matrix</em>, <span class="math inline">\(\dataMatrix\)</span>, where each <em>row</em> of <span class="math inline">\(\dataMatrix\)</span> contains the observation for one individual. To emphasize that <span class="math inline">\(\dataVector\)</span> is a vector derived from a row of <span class="math inline">\(\dataMatrix\)</span> we represent the observation of the features associated with the <span class="math inline">\(i\)</span>th individual by <span class="math inline">\(\dataVector_{i, :}\)</span>, and place each individual in our data matrix, <span class="math display">\[
\dataMatrix
= \begin{bmatrix} \dataVector_{1, :}^\top \\ \dataVector_{2, :}^\top \\ \vdots \\
\dataVector_{n, :}^\top\end{bmatrix},
\]</span> where we have <span class="math inline">\(n\)</span> data points. Our data matrix therefore has <span class="math inline">\(n\)</span> rows and <span class="math inline">\(p\)</span> columns. The point to notice here is that each data obsesrvation appears as a row vector in the design matrix (thus the transpose operation inside the brackets). Our prediction functions are now actually a <em>matrix value</em> function, <span class="math display">\[
\mathbf{F} = \latentMatrix\mappingMatrix^\top,
\]</span> where for each matrix the data points are in the rows and the data features are in the columns. This implies that if we have <span class="math inline">\(q\)</span> inputs to the function we have <span class="math inline">\(\mathbf{F}\in \Re^{n\times p}\)</span>, <span class="math inline">\(\mappingMatrix \in \Re^{p \times q}\)</span> and <span class="math inline">\(\latentMatrix \in \Re^{n\times q}\)</span>.</p>
</section>
<section id="assignment-question-1" class="slide level3">
<h3>Assignment Question 1</h3>
<p>Show that, given all the definitions above, if, <span class="math display">\[
\mathbf{F} = \latentMatrix\mappingMatrix^\top
\]</span> and the elements of the vector valued function <span class="math inline">\(\mathbf{F}\)</span> are given by <span class="math display">\[
f_{i, j} = f_j(\mathbf{x}_{i, :}),
\]</span> where <span class="math inline">\(\mathbf{x}_{i, :}\)</span> is the <span class="math inline">\(i\)</span>th row of the latent variables, <span class="math inline">\(\latentMatrix\)</span>, then show that <span class="math display">\[
f_j(\mathbf{x}_{i, :}) = \mathbf{w}_{j, :}^\top
\mathbf{x}_{i, :}
\]</span></p>
<p><em>10 marks</em></p>
<h4 id="question-1-answer">Question 1 Answer</h4>
<p>Write your answer to the question in this box.</p>
</section>
<section><section id="latent-variables" class="titleslide slide level2"><h2>Latent Variables</h2></section></section>
<section><section id="section" class="titleslide slide level2"><h2></h2></section><section id="positive-definite" class="slide level3">
<h3>Positive Definite</h3>
<p>We are interested in the case where <span class="math inline">\(\mathbf{A}\)</span> is a covariance matrix, which implies it is <em>positive definite</em>. A positive definite matrix is one for which the inner product, <span class="math display">\[
\mathbf{w}^\top \mathbf{C}\mathbf{w}
\]</span> is positive for <em>all</em> values of the vector <span class="math inline">\(\mathbf{w}\)</span> other than the zero vector. One way of creating a positive definite matrix is to assume that the symmetric and positive definite matrix <span class="math inline">\(\mathbf{C}\in \Re^{p\times p}\)</span> is factorised into, <span class="math inline">\(\mathbf{A}in \Re^{p\times p}\)</span>, a <em>full rank</em> matrix, so that <span class="math display">\[
\mathbf{C} = \mathbf{A}^\top
\mathbf{A}.
\]</span> This ensures that <span class="math inline">\(\mathbf{C}\)</span> must be positive definite because <span class="math display">\[
\mathbf{w}^\top \mathbf{C}\mathbf{w}=\mathbf{w}^\top
\mathbf{A}^\top\mathbf{A}\mathbf{w} 
\]</span> and if we now define a new <em>vector</em> <span class="math inline">\(\mathbf{b}\)</span> as <span class="math display">\[
\mathbf{b} = \mathbf{A}\mathbf{w}
\]</span> we can now rewrite as <span class="math display">\[
\mathbf{w}^\top \mathbf{C}\mathbf{w} = \mathbf{b}^\top\mathbf{b} = \sum_{i}
b_i^2
\]</span> which, since it is a sum of squares, is positive or zero. The constraint that <span class="math inline">\(\mathbf{A}\)</span> must be <em>full rank</em> ensures that there is no vector <span class="math inline">\(\mathbf{w}\)</span>, other than the zero vector, which causes the vector <span class="math inline">\(\mathbf{b}\)</span> to be all zeros.</p>
</section><section id="assignment-question-2" class="slide level3">
<h3>Assignment Question 2</h3>
<p>If <span class="math inline">\(\mathbf{C}=\mathbf{A}^\top \mathbf{A}\)</span> then express <span class="math inline">\(c_{i,j}\)</span>, the value of the element at the <span class="math inline">\(i\)</span>th row and the <span class="math inline">\(j\)</span>th column of <span class="math inline">\(\mathbf{C}\)</span>, in terms of the columns of <span class="math inline">\(\mathbf{A}\)</span>. Use this to show that (i) the matrix is symmetric and (ii) the matrix has positive elements along its diagonal.</p>
<p><em>15 marks</em></p>
<h4 id="question-2-answer">Question 2 Answer</h4>
<p>Write your answer to the question in this box.</p>
<!-- SECTION Eigenvectors of a Symmetric Matric -->
<p>Symmetric matrices have <em>orthonormal</em> eigenvectors. This means that <span class="math inline">\(\mathbf{U}\)</span> is an <a href="http://en.wikipedia.org/wiki/Orthogonal_matrix">orthogonal matrix</a>, <span class="math inline">\(\mathbf{U}^\top\mathbf{U} = \mathbf{I}\)</span>. This implies that <span class="math inline">\(\mathbf{u}_{:, i} ^\top \mathbf{u}_{:, j}\)</span> is equal to 0 if <span class="math inline">\(i\neq j\)</span> and 1 if <span class="math inline">\(i=j\)</span>.</p>
<!-- SECTION Probabilistic PCA -->
<p>In 1997 <a href="http://research.microsoft.com/pubs/67218/bishop-ppca-jrss.pdf">Tipping and Bishop</a> and <a href="https://www.cs.nyu.edu/~roweis/papers/empca.pdf">Roweis</a> independently revisited Hotelling’s model and considered the case where the noise variance was finite, but <em>shared</em> across all output dimensons. Their model can be thought of as a factor analysis where <span class="math display">\[
\boldsymbol{\Sigma} = \sigma^2 \mathbf{I}.
\]</span> This leads to a marginal likelihood of the form <span class="math display">\[
p(\dataMatrix|\mappingMatrix, \sigma^2)
= \prod_{i=1}^n\mathcal{N}(\dataVector_{i, :} | \mathbf{0},
\mappingMatrix\mappingMatrix^\top + \sigma^2 \mathbf{I})
\]</span> where the limit of <span class="math inline">\(\sigma^2\rightarrow 0\)</span> is <em>not</em> taken. This defines a proper probabilistic model. Tippping and Bishop then went on to prove that the <em>maximum likelihood</em> solution of this model with respect to <span class="math inline">\(\mappingMatrix\)</span> is given by an eigenvalue problem. In the probabilistic PCA case the eigenvalues and eigenvectors are given as follows. <span class="math display">\[
\mappingMatrix = \mathbf{U}\mathbf{L} \mathbf{R}^\top
\]</span> where <span class="math inline">\(\mathbf{U}\)</span> is the eigenvectors of the empirical covariance matrix <span class="math display">\[
\mathbf{S} = \sum_{i=1}^n (\dataVector_{i, :} - \boldsymbol{\mu})(\dataVector_{i,
:} - \boldsymbol{\mu})^\top,
\]</span> which can be written <span class="math inline">\(\mathbf{S} = \frac{1}{\numData} \dataMatrix^\top\dataMatrix\)</span> if the data is zero mean. The matrix <span class="math inline">\(\mathbf{L}\)</span> is diagonal and is dependent on the <em>eigenvalues</em> of <span class="math inline">\(\mathbf{S}\)</span>, <span class="math inline">\(\boldsymbol{\Lambda}\)</span>. If the <span class="math inline">\(i\)</span>th diagonal element of this matrix is given by <span class="math inline">\(\lambda_i\)</span> then the corresponding element of <span class="math inline">\(\mathbf{L}\)</span> is <span class="math display">\[
\ell_i =
\sqrt{\lambda_i - \sigma^2}
\]</span> where <span class="math inline">\(\sigma^2\)</span> is the noise variance. Note that if <span class="math inline">\(\sigma^2\)</span> is larger than any particular eigenvalue, then that eigenvalue (along with its corresponding eigenvector) is <em>discarded</em> from the solution.</p>
</section><section id="python-implementation-of-probabilistic-pca" class="slide level3">
<h3>Python Implementation of Probabilistic PCA</h3>
<p>We will now implement this algorithm in python.</p>
<p>In practice we may not wish to compute the eigenvectors of the covariance matrix directly. This is because it requires us to estimate the covariance, which involves a sum of squares term, before estimating the eigenvectors. We can estimate the eigenvectors directly either through <a href="http://en.wikipedia.org/wiki/QR_decomposition">QR decomposition</a> or <a href="http://en.wikipedia.org/wiki/Singular_value_decomposition">singular value decomposition</a>. We saw a similar issue arise when <a href="./week3.ipynb">computing the weights in a regression problem</a>, where we also wished to avoid computation of <span class="math inline">\(\latentMatrix^\top\latentMatrix\)</span> (or in the case of <a href="./week4.ipynb">nonlinear regression with basis functions</a> <span class="math inline">\(\boldsymbol{\Phi}^\top\boldsymbol{\Phi}\)</span>).</p>
<!-- SECTION Posterior for Principal Component Analysis -->
<p>Under the latent variable model justification for principal component analysis, we are normally interested in inferring something about the latent variables given the data. This is the distribution, <span class="math display">\[
p(\mathbf{x}_{i, :} | \dataVector_{i, :})
\]</span> for any given data point. Determining this density turns out to be very similar to the approach for determining the Bayesian posterior of <span class="math inline">\(\mathbf{w}\)</span> in Bayesian linear regression, only this time we place the prior density over <span class="math inline">\(\mathbf{x}_{i, :}\)</span> instead of <span class="math inline">\(\mathbf{w}\)</span>. The posterior is proportional to the joint density as follows, <span class="math display">\[
p(\mathbf{x}_{i, :} | \dataVector_{i, :}) \propto p(\dataVector_{i,
:}|\mappingMatrix, \mathbf{x}_{i, :}, \sigma^2) p(\mathbf{x}_{i, :})
\]</span> And as in the Bayesian linear regression case we first consider the log posterior, <span class="math display">\[
\log
p(\mathbf{x}_{i, :} | \dataVector_{i, :}) = \log p(\dataVector_{i, :}|\mappingMatrix,
\mathbf{x}_{i, :}, \sigma^2) + \log p(\mathbf{x}_{i, :}) + \text{const}
\]</span> where the constant is not dependent on <span class="math inline">\(\mathbf{x}\)</span>. As before we collect the quadratic terms in <span class="math inline">\(\mathbf{x}_{i, :}\)</span> and we assemble them into a Gaussian density over <span class="math inline">\(\mathbf{x}\)</span>. <span class="math display">\[
\log p(\mathbf{x}_{i, :} | \dataVector_{i, :}) =
-\frac{1}{2\sigma^2} (\dataVector_{i, :} - \mappingMatrix\mathbf{x}_{i,
:})^\top(\dataVector_{i, :} - \mappingMatrix\mathbf{x}_{i, :}) - \frac{1}{2}
\mathbf{x}_{i, :}^\top \mathbf{x}_{i, :} + \text{const}
\]</span></p>
</section><section id="assignment-question-3" class="slide level3">
<h3>Assignment Question 3</h3>
<p>Multiply out the terms in the brackets. Then collect the quadratic term and the linear terms together. Show that the posterior has the form <span class="math display">\[
\mathbf{x}_{i, :} | \mappingMatrix \sim \mathcal{N}( \boldsymbol{\mu}_x,
\mathbf{C}_x)
\]</span> where <span class="math display">\[
\mathbf{C}_x = \left(\sigma^{-2}
\mappingMatrix^\top\mappingMatrix + \mathbf{I}\right)^{-1}
\]</span> and <span class="math display">\[
\boldsymbol{\mu}_x
= \mathbf{C}_x \sigma^{-2}\mappingMatrix^\top \dataVector_{i, :} 
\]</span> Compare this to the posterior for the Bayesian linear regression from last week, do they have similar forms? What matches and what differs?</p>
<p><em>30 marks</em></p>
<h4 id="question-3-answer">Question 3 Answer</h4>
<p>Write your answer to the question in this box. Use latex to write your derivations, show each step of your work.</p>
</section><section id="python-implementation-of-the-posterior" class="slide level3">
<h3>Python Implementation of the Posterior</h3>
<p>Now let’s implement the system in code.</p>
</section><section id="assignment-question-4" class="slide level3">
<h3>Assignment Question 4</h3>
<p>Use the values for <span class="math inline">\(\mappingMatrix\)</span> and <span class="math inline">\(\sigma^2\)</span> you have computed, along with the data set <span class="math inline">\(\dataMatrix\)</span> to compute the posterior density over <span class="math inline">\(\latentMatrix\)</span>. Write a function of the form}python mu_x, C_x = posterior(Y, W, sigma2)} where <code>mu_x</code> and <code>C_x</code> are the posterior mean and posterior covariance for the given <span class="math inline">\(\dataMatrix\)</span>.</p>
<p>Don’t forget to subtract the mean of the data <code>Y</code> inside your function before computing the posterior: remember we assumed at the beginning of our analysis that the data had been centred (i.e. the mean was removed).</p>
<p><em>20 marks</em></p>
</section></section>
<section><section id="numerically-stable-and-efficient-version" class="titleslide slide level1"><h1>Numerically Stable and Efficient Version</h1></section><section id="solution-for-mappingmatrix" class="slide level3">
<h3>Solution for <span class="math inline">\(\mappingMatrix\)</span></h3>
<p>Given the singular value decomposition of <span class="math inline">\(\dataMatrix\)</span> then we have <span class="math display">\[
\mappingMatrix =
\mathbf{U}\mathbf{L}\mathbf{R}^\top
\]</span> where <span class="math inline">\(\mathbf{R}\)</span> is an arbitrary rotation matrix. This implies that the posterior is given by <span class="math display">\[
\mathbf{C}_x =
\left[\sigma^{-2}\mathbf{R}\mathbf{L}^2\mathbf{R}^\top + \mathbf{I}\right]^{-1}
\]</span> because <span class="math inline">\(\mathbf{U}^\top \mathbf{U} = \mathbf{I}\)</span>. Since, by convention, we normally take <span class="math inline">\(\mathbf{R} = \mathbf{I}\)</span> to ensure that the principal components are orthonormal we can write <span class="math display">\[
\mathbf{C}_x = \left[\sigma^{-2}\mathbf{L}^2 +
\mathbf{I}\right]^{-1}
\]</span> which implies that <span class="math inline">\(\mathbf{C}_x\)</span> is actually diagonal with elements given by <span class="math display">\[
c_i = \frac{\sigma^2}{\sigma^2 + \ell^2_i}
\]</span> and allows us to write <span class="math display">\[
\boldsymbol{\mu}_x = [\mathbf{L}^2 + \sigma^2
\mathbf{I}]^{-1} \mathbf{L} \mathbf{U}^\top \dataVector_{i, :}
\]</span> <span class="math display">\[
\boldsymbol{\mu}_x = \mathbf{D}\mathbf{U}^\top \dataVector_{i, :}
\]</span> where <span class="math inline">\(\mathbf{D}\)</span> is a diagonal matrix with diagonal elements given by <span class="math inline">\(d_{i} = \frac{\ell_i}{\sigma^2 + \ell_i^2}\)</span>.</p>
<!-- SECTION Examples -->
<p>For our first example we’ll consider some motion capture data of a man breaking into a run. <a href="http://en.wikipedia.org/wiki/Motion_capture">Motion capture data</a> involves capturing a 3-d point cloud to represent a character, often by an underlying skeleton. For this data set, from Ohio State University, we have 54 frame of motion capture, each frame containing 102 values, which are the 3-d locations of 34 different points from the subjects skeleton.</p>
<p>Once the data is loaded in we can examine the first two principal components as follows,</p>
<p>Here because the data is a time course, we have connected points that are neighbouring in time. This highlights the form of the run, which involves 3 paces. This projects in our low dimensional space to 3 loops. We can examin how much residual variance there is in the system by looking at <code>sigma2</code>.</p>
<!-- SECTION Robot Navigation Example -->
<p>In the next example we will load in data from a robot navigation problem. The data consists of wireless access point strengths as recorded by a robot performing a loop around the University of Washington’s Computer Science department in Seattle. The robot records all the wireless access points it can cache and stores their signal strength.</p>
<p>There are 215 observations of 30 different access points. In this case the model is suggesting that the access point signal strength should be linearly dependent on the location in the map. In other words we are expecting the access point strength for the <span class="math inline">\(j\)</span>th access point at robot position <span class="math inline">\(x_{i, :}\)</span> to be represented by <span class="math inline">\(y_{i, j} = \mathbf{w}_{j, :}^\top \mathbf{x}_{i, :} + \epsilon_{i,j}\)</span>.</p>
<!-- SECTION Relationship to Matrix Factorization -->
<p>We can use the robot naviation example to realise that PCA (and factor analysis) are very reminiscient of the <a href="./week2.ipynb"><em>matrix factorization</em> example</a> that we used for introducing objective functions. In that system we used slightly different notation, <span class="math inline">\(\mathbf{u}_{i, :}\)</span> for <em>user</em> location in our metaphorical library and <span class="math inline">\(\mathbf{v}_{j, :}\)</span> for <em>item</em> location in our metaphorical library. To see how these systems are somewhat analagous, now let us think about the user as the robot and the items as the wifi access points. We can plot the relative location of both. This process is known as “SLAM”: simultaneous <em>localisation</em> and <em>mapping</em>. A latent variable model of the type we have developed is one way of performing SLAM. We have an estimate of the <em>landmarks</em> in the system (in this case WIFI access points) and we have an estimate of the robot position. These are analagous to the estimate of the user’s position and the estimate of the items positions in the library. In the matrix factorisation example users are informing us what items they are ‘close’ to by expressing their preferences, in the robot localization example the robot is informing us what access point it is close to by measuring signal strength.</p>
<p>From a personal perspective, I find this analogy quite comforting. I think it is very arguable that one of the mechanisms through which we (as humans) may have developed higher reasoning is through the need to navigate around our environment, identifying landmarks and associating them with our search for food. If such a system were to exist, the idea that it could be readily adapted to other domains such as categorising the nature of the different foodstuffs we were able to forage is intriguing.</p>
<p>From an algorithmic perspective, we also can now realise that matrix factorization and latent variable modelling are effectively the same thing. The only difference is the objective function and our probabilistic (or lack of probabilistic) treatment of the variables. But the prediction functoin for both systems, <span class="math display">\[
f_{i, j} =
\mathbf{u}_{i, :}^\top \mathbf{v}_{j, :} 
\]</span> for matrix factorization or <span class="math display">\[
f_{i, j} = \mathbf{x}_{i, :}^\top \mathbf{w}_{j, :} 
\]</span> for probabilistic PCA and factor analysis are the same.</p>
<!-- SECTION Other Interpretations of PCA: Separating Model and Algorithm -->
<p>Since Hotelling first introduced his perspective on factor analysis as PCA there has been somewhat of a conflation of the idea of the model underlying PCA (for which it was very clear that Hotelling was inspired by Factor Analysis) and the algorithm that is used to fit that model: the eigenvalues and eigenvectors of the covariance matrix. The eigenvectors of an ellipsoid have been known since the middle of the 19th century as the principal axes of the elipsoid, and they arise through the following additional ideas: seeking the orthogonal directions of <em>maximum variance</em> in a dataset. Pearson in 1901 arrived at the same algorithm driven by a desire to seek a <em>symmetric regression</em> between two covariate/response variables <span class="math inline">\(x\)</span> and <span class="math inline">\(y\)</span>. He is, therefore, often credited with the invention of principal component analysis, but to me this seems disengenous. His aim was very different from Hotellings, it was just happened that the optimal solution for his model was coincident with that of Hotelling. The approach is also known as the <a href="http://en.wikipedia.org/wiki/Karhunen%E2%80%93Lo%C3%A8ve_theorem">Karhunen Loeve Transform</a> in stochastic process theory and in classical multidimensional scaling the same operation can be shown to be minimising a particular objective function based on interpoint distances in the data and the latent space (see the section on Classical Multidimensional Scaling in <a href="http://store.elsevier.com/Multivariate-Analysis/Kanti-%20Mardia/isbn-9780124712522/">Mardia, Kent and Bibby</a>). One of my own contributions to machine learning was deriving yet another model whose linear variant was solved by finding the principal subspace of the covariance matrix (an approach I termed dual probabilistic PCA or probabilistic principal coordinate analysis). Finally, the approach is sometimes referred to simply as singular value decomposition (SVD). The singular value decomposition of a data set has the following form, <span class="math display">\[
\dataMatrix = \mathbf{V} \boldsymbol{\Lambda} \mathbf{U}^\top
\]</span> where <span class="math inline">\(\mathbf{V}\in\Re^{n\times n}\)</span> and <span class="math inline">\(\mathbf{U}^\in \Re^{p\times p}\)</span> are square orthogonal matrices and <span class="math inline">\(\mathbf{\Lambda}^{n \times p}\)</span> is zero apart from its first <span class="math inline">\(p\)</span> diagonal entries. Singularvalue decomposition gives a diagonalisation of the covariance matrix, because under the SVD we have <span class="math display">\[
\dataMatrix^\top\dataMatrix =
\mathbf{U}\boldsymbol{\Lambda}\mathbf{V}^\top\mathbf{V} \boldsymbol{\Lambda}
\mathbf{U}^\top = \mathbf{U}\boldsymbol{\Lambda}^2 \mathbf{U}^\top
\]</span> where <span class="math inline">\(\boldsymbol{\Lambda}^2\)</span> is now the eigenvalues of the covariane matrix and <span class="math inline">\(\mathbf{U}\)</span> are the eigenvectors. So performing the SVD can simply be seen as another approach to determining the principal components.</p>
</section><section id="separating-model-and-algorithm" class="slide level3">
<h3>Separating Model and Algorithm</h3>
<p>I’ve given a fair amount of personal thought to this situation and my own opinion that this confusion about method arises because of a conflation of model and algorithm. The model of Hotelling, that which he termed principal component analysis, was really a variant of factor analysis, and it was unfortunate that he chose to rename it. However, the algorithm he derived was a very convenient way of optimising a (simplified) factor analysis, and it’s therefore become very popular. The algorithm is also the optimal solution for many other models of the data, even some which might seem initally to be unrelated (e.g. seeking directions of maximum variance). It is only through the mathematics of this linear system (which also contains some intersting symmetries) that all these ides become related. However, as soon as we choose to non-linearise the system (e.g. through basis functions) we find that each of the non-linear intepretations we can derive for the different models each leads to a very different algorithm (if such an algorithm is possible). For example <a href="http://web.stanford.edu/~hastie/Papers/Principal_Curves.pdf">principal curves</a> of Hastie and Stuezle attempt to non-linearise the maximum variance interpretation, <a href="http://en.wikipedia.org/wiki/Kernel_principal_component_analysis">kernel PCA</a> of Schoelkopf, Smola and Mueller uses basis functions to form the eigenvalue problem in a nonlinear space, and my own work in this area <a href="http://jmlr.org/papers/volume6/lawrence05a/lawrence05a.pdf">non-linearises the dual probabilistic PCA</a>.</p>
<p>My conclusion is that when you are doing machine learning you should always have it clear in your mind what your <em>model</em> is and what your <em>algorithm</em> is. You can recognise your model because it normally contains a prediction function and an objective function. The algorithm on the other hand is the sequence of steps you implement on the computer to solve for the parameters of this model. For efficient implementation, we often modify our model to allow for faster algorithms, and this is a perfectly valid pragmatist’s approach, so conflation of model and algorithm is not always a bad thing. But for clarity of thinking and understanding it is necessary to maintain the separation and to maintain a handle on when and why we perform the conflation.</p>
<!-- SECTION PCA in Practice -->
<p>Principal component analysis is so effective in practice that there has almost developed a mini-industry in renaming the method itself (which is ironic, given its origin). In particular <a href="http://en.wikipedia.org/wiki/Latent_semantic_indexing">Latent Semantic Indexing</a> in text processing is simply PCA on a particular representation of the term frequencies of the document. There is a particular fad to rename the eigenvectors after the nature of the data you are examining, perhaps initially triggered by <a href="http://www.face-rec.org/algorithms/PCA/jcn.pdf">Turk and Pentland’s</a> paper on eigenfaces, but also with <a href="https://wiki.inf.ed.ac.uk/twiki/pub/CSTR/ListenSemester1_2007_8/kuhn-%20junqua-eigenvoice-icslp1998.pdf">eigenvoices</a> and <a href="http://www.biomedcentral.com/1752-0509/1/54">eigengenes</a>. This seems to be an instantiation of a wider, and hopefully subconcious, tendency in academia to attempt to differentiate one idea from the same idea in related fields in order to emphasise the novelty. The unfortunate result is somewhat of a confusing literature for relatively simple model. My recommendations would be as follows. If you have multivariate data, applying some form of principal component would seem to be a very good idea as a first step. Even if you intend to later perform classification or regression on your data, it can give you understanding of the structure of the underlying data and help you to develop your intuitions about the nature of your data. Intelligent plotting and interaction with your data is always a good think, and for high dimensional data that means that you need some way of visualisation, PCA is typically a good starting point.</p>
<!-- SECTION Marginal Likelihood -->
<p>We have developed the posterior density over the latent variables given the data and the parameters, and due to symmetries in the underlying prediction function, it has a very similar form to its sister density, the posterior of the weights given the data from Bayesian regression. Two key differences are as follows. If we were to do a Bayesian multiple output regression we would find that the marginal likelihood of the data is independent across the features and correlated across the data, <span class="math display">\[
p(\dataMatrix|\latentMatrix)
= \prod_{j=1}^p \mathcal{N}(\dataVector_{:, j}|\mathbf{0},
\alpha\latentMatrix\latentMatrix^\top + \sigma^2 \mathbf{I})
\]</span> where <span class="math inline">\(\dataVector_{:, j}\)</span> is a column of the data matrix and the independence is across the <em>features</em>, in probabilistic PCA the marginal likelihood has the form, <span class="math display">\[
p(\dataMatrix|\mappingMatrix) = \prod_{i=1}^n \mathcal{N}(\dataVector_{i,
:}|\mathbf{0}, \mappingMatrix\mappingMatrix^\top + \sigma^2 \mathbf{I})
\]</span> where <span class="math inline">\(\dataVector_{i, :}\)</span> is a row of the data matrix <span class="math inline">\(\dataMatrix\)</span> and the independence is across the data points.</p>
<!-- SECTION Computation of the Log Likelihood -->
<p>The quality of the model can be assessed using the log likelihood of this Gaussian form. <span class="math display">\[
\log p(\dataMatrix|\mappingMatrix) = -\frac{\numData}{2} \log \left|
\mappingMatrix\mappingMatrix^\top + \sigma^2 \mathbf{I} \right| -\frac{1}{2}
\sum_{i=1}^n \dataVector_{i, :}^\top \left(\mappingMatrix\mappingMatrix^\top + \sigma^2
\mathbf{I}\right)^{-1} \dataVector_{i, :} +\text{const}
\]</span> but this can be computed more rapidly by exploiting the low rank form of the covariance covariance, <span class="math inline">\(\mathbf{C} = \mappingMatrix\mappingMatrix^\top + \sigma^2 \mathbf{I}\)</span> and the fact that <span class="math inline">\(\mappingMatrix = \mathbf{U}\mathbf{L}\mathbf{R}^\top\)</span>. Specifically, we first use the decomposition of <span class="math inline">\(\mappingMatrix\)</span> to write: <span class="math display">\[
-\frac{\numData}{2} \log \left| \mappingMatrix\mappingMatrix^\top + \sigma^2 \mathbf{I} \right|
= -\frac{\numData}{2} \sum_{i=1}^q \log (\ell_i^2 + \sigma^2) - \frac{n(p-q)}{2}\log
\sigma^2 ,
\]</span> where <span class="math inline">\(\ell_i\)</span> is the <span class="math inline">\(i\)</span>th diagonal element of <span class="math inline">\(\mathbf{L}\)</span>. Next, we use the <a href="http://en.wikipedia.org/wiki/Woodbury_matrix_identity">Woodbury matrix identity</a> which allows us to write the inverse as a quantity which contains another inverse in a smaller matrix: <span class="math display">\[
(\sigma^2 \mathbf{I} + \mappingMatrix\mappingMatrix^\top)^{-1} =
\sigma^{-2}\mathbf{I}-\sigma^{-4}\mappingMatrix{\underbrace{(\mathbf{I}+\sigma^{-2}\mappingMatrix^\top\mappingMatrix)}_{\mathbf{C}_x}}^{-1}\mappingMatrix^\top
\]</span> So, it turns out that the original inversion of the <span class="math inline">\(p \times p\)</span> matrix can be done by forming a quantity which contains the inversion of a <span class="math inline">\(q \times q\)</span> matrix which, moreover, turns out to be the quantity <span class="math inline">\(\mathbf{C}_x\)</span> of the posterior.</p>
<p>Now, we put everything together to obtain: <span class="math display">\[
\log
p(\dataMatrix|\mappingMatrix) = -\frac{\numData}{2} \sum_{i=1}^q \log (\ell_i^2 + \sigma^2)
- \frac{n(p-q)}{2}\log \sigma^2 - \frac{1}{2} \text{tr} \left( \dataMatrix^\top
\left( \sigma^{-2}\mathbf{I}-\sigma^{-4}\mappingMatrix \mathbf{C}_x \mappingMatrix^\top
\right) \dataMatrix  \right) + \text{const}
\]</span> where we used the fact that a scalar sum can be written as <span class="math inline">\(\sum_{i=1}^n \dataVector_{i,:}^\top \kernelMatrix \dataVector_{i,:} = \text{tr}\left( \dataMatrix^\top \kernelMatrix \dataMatrix \right)\)</span>, for any matrix <span class="math inline">\(\kernelMatrix\)</span> of appropriate dimensions. We now use the properties of the trace <span class="math inline">\(\text{tr}(\mathbf{A}+\mathbf{B})=\text{tr}(\mathbf{A})+\text{tr}(\mathbf{B})\)</span> and <span class="math inline">\(\text{tr}(c \mathbf{A}) = c \text{tr}(\mathbf{A})\)</span>, where <span class="math inline">\(c\)</span> is a scalar and <span class="math inline">\(\mathbf{A},\mathbf{B}\)</span> matrices of compatible sizes. Therefore, the final log likelihood takes the form: <span class="math display">\[
\log p(\dataMatrix|\mappingMatrix) = -\frac{\numData}{2}
\sum_{i=1}^q \log (\ell_i^2 + \sigma^2) - \frac{\numData(p-q)}{2}\log \sigma^2 -
\frac{\sigma^{-2}}{2} \text{tr}(\dataMatrix^\top \dataMatrix)
+\frac{\sigma^{-4}}{2} \text{tr}(\mathbf{B}\mathbf{C}_x\mathbf{B}^\top) +
\text{const}
\]</span> where we also defined <span class="math inline">\(\mathbf{B}=\dataMatrix^\top\mappingMatrix\)</span>. Finally, notice that <span class="math inline">\(\text{tr}(\dataMatrix\dataMatrix^\top)=\text{tr}(\dataMatrix^\top\dataMatrix)\)</span> can be computed faster as the sum of all the elements of <span class="math inline">\(\dataMatrix\circ\dataMatrix\)</span>, where <span class="math inline">\(\circ\)</span> denotes the element-wise (or <a href="http://en.wikipedia.org/wiki/Hadamard_product_(matrices)">Hadamard</a> product.</p>
</section><section id="reconstruction-of-the-data" class="slide level3">
<h3>Reconstruction of the Data</h3>
<p>Given any posterior projection of a data point, we can replot the original data as a function of the input space.</p>
<p>We will now try to reconstruct the motion capture figure form some different places in the latent plot.</p>
</section><section id="assignment-question-5" class="slide level3">
<h3>Assignment Question 5</h3>
</section><section id="other-data-sets-to-explore" class="slide level3">
<h3>Other Data Sets to Explore</h3>
<p>Below there are a few other data sets from <code>pods</code> you might want to explore with PCA. Both of them have <span class="math inline">\(p\)</span>&gt;<span class="math inline">\(n\)</span> so you need to consider how to do the larger eigenvalue probleme efficiently without large demands on computer memory.</p>
<p>The data is actually quite high dimensional, and solving the eigenvalue problem in the high dimensional space can take some time. At this point we turn to a neat trick, you don’t have to solve the full eigenvalue problem in the <span class="math inline">\(\dataDim\times \dataDim\)</span> covariance, you can choose instead to solve the related eigenvalue problem in the <span class="math inline">\(\numData \times \numData\)</span> space, and in this case <span class="math inline">\(\numData=200\)</span> which is much smaller than <span class="math inline">\(\dataDim\)</span>.</p>
<p>The original eigenvalue problem has the form <span class="math display">\[
\dataMatrix^\top\dataMatrix \mathbf{U} = \mathbf{U}\boldsymbol{\Lambda}
\]</span> But if we premultiply by <span class="math inline">\(\dataMatrix\)</span> then we can solve, <span class="math display">\[
\dataMatrix\dataMatrix^\top\dataMatrix \mathbf{U} = \dataMatrix\mathbf{U}\boldsymbol{\Lambda}
\]</span> but it turns out that we can write <span class="math display">\[
\mathbf{U}^\prime = \dataMatrix \mathbf{U} \Lambda^{\frac{1}{2}}
\]</span> where <span class="math inline">\(\mathbf{U}^\prime\)</span> is an orthorormal matrix because <span class="math display">\[
\left.\mathbf{U}^\prime\right.^\top\mathbf{U}^\prime = \Lambda^{-\frac{1}{2}}\mathbf{U}\dataMatrix^\top\dataMatrix \mathbf{U} \Lambda^{-\frac{1}{2}}
\]</span> and since <span class="math inline">\(\mathbf{U}\)</span> diagonalises <span class="math inline">\(\dataMatrix^\top\dataMatrix\)</span>, <span class="math display">\[
\mathbf{U}\dataMatrix^\top\dataMatrix \mathbf{U} = \Lambda
\]</span> then <span class="math display">\[
\left.\mathbf{U}^\prime\right.^\top\mathbf{U}^\prime = \mathbf{I}
\]</span></p>
</section><section id="olivetti-faces" class="slide level3">
<h3>Olivetti Faces</h3>
<p>You too can create your own eigenfaces. In this example we load in the ‘Olivetti Face’ data set, a small data set of 200 faces from the <a href="http://en.wikipedia.org/wiki/Olivetti_Research_Laboratory">Olivetti Research Laboratory</a>. Below we load in the data and display an image of the second face in the data set (i.e., indexed by 1).</p>
<p>Note that to display the face we had to reshape the appropriate row of the data matrix. This is because the images are turned into vectors by stacking columns of the image on top of each other to form a vector. The operation</p>
<p><code>im = np.reshape(Y[1, :].flatten(), (64, 64)).T}</code></p>
<p>recovers the original image into a matrix <code>im</code> by using the <code>np.reshape</code> function to return the vector to a matrix.</p>
</section><section id="visualizing-the-eigenvectors" class="slide level3">
<h3>Visualizing the Eigenvectors</h3>
<p>Each retained eigenvector is stored in the <span class="math inline">\(j\)</span>th column of <span class="math inline">\(\mathbf{U}\)</span>. Each of these eigenvectors is associated with particular directions of variation in the original data. Principal component analysis implies that we can reconstruct any face by using a weighted sum of these eigenvectors where the weights for each face are given by the relevant vector of the latent variables, <span class="math inline">\(\mathbf{x}_{i, :}\)</span> and the diagonal elements of the matrix <span class="math inline">\(\mathbf{L}\)</span>. We can visualize the eigenvectors <span class="math inline">\(\mathbf{U}\)</span> as images by performing the same reshape operation we used to recover the image associated with a data point above. Below we do this for the first nine eigenvectors of the Olivetti Faces data.</p>
</section><section id="reconstruction" class="slide level3">
<h3>Reconstruction</h3>
<p>We can now attempt to reconstruct a given training point from these eigenvectors. As we mentioned above, the reconstruction is dependent on the value of the latent variable and the weights from the matrix <span class="math inline">\(\mathbf{L}\)</span>. First let’s compute the value of the latent variables for the point we want to construct. Then we’ll use them to compute the weightings of each of the eigenvectors.</p>
<p>This vector of reconstruction weights is applied to the ‘template images’ given by the eigenvectors to give us our reconstruction. Below we weight these templates and combine to form the reconstructed image, and show the comparison to the original image.</p>
<p>The quality of the reconstruction is a bit blurry, it can be improved by increasing the number of template images used (i.e. increasing the <em>latent dimensionality</em>).</p>
</section><section id="gene-expression" class="slide level3">
<h3>Gene Expression</h3>
<p>Each of the cells in your body stores your entire genetic code in your DNA, but at any one moment it is only ‘expressing’ a small portion of that code. Your cells are mainly constructed of protein, and these proteins are manufactured by first transcribing the DNA to RNA and then translating the RNA to protein. If the DNA is the cells hard drive, then one role of the RNA is to act like a cache of data that is being read from the hard drive at any one time. Gene expression arrays allow us to measure the quantity of different types of RNA in the cell, effectively analyzing what’s in the cache (although we have to destroy the cell or the tissue to access it). A gene expression experiment often consists of a time course or a series of experiments that characterise the gene expression of cells at any given time.</p>
<p>We will now load in one of the earliest gene expression data sets from a <a href="http://www.ncbi.nlm.nih.gov/pubmed/9843569">1998 paper by Spellman et al.</a>, it consists of gene expression measurements of over six thousand genes in a range of conditions for brewer’s yeast. The experiment was designed for understanding the cell cycle of the genes. The expectation is that there should be oscillating signals inside the cell.</p>
<p>First we extract the principale components of the gene expression.</p>
<p>Now, looking through, we find that there is indeed a latent variable that appears to oscilate at approximately the right frequency. The 4th latent dimension (<code>index=3</code>) can be plotted across the time course as follows.</p>
<p>To reveal an oscillating shape. We can see which genes correspond to this shape by examining the associated column of <span class="math inline">\(\mathbf{U}\)</span>. Let’s augment our data matrix with this score.</p>
<p>We can look up the first three genes in this list which now ranks the genes according to how strongly they respond to the fourth latent dimension. <a href="http://www.ncbi.nlm.nih.gov/gene/">The NCBI gene database</a> allows us to search for the function of these genes. Looking at the function of the four genes that respond most strongly to the third latent variable they are all genes that encode <a href="http://en.wikipedia.org/wiki/Histone">histone</a> proteins. The histone is thesupport scaffold for the DNA that ensures it folds correctly within the cell creating the nucleosomes. It seems to make sense that production of histone proteins should be strongly correlated with the cell cycle, as when the cell divides it needs to create a large number of histone proteins for creating the duplicated nucleosomes. The relevant links giving the descriptions of each gene given here: <a href="http://www.ncbi.nlm.nih.gov/gene/851810">YDR224C</a>, <a href="http://www.ncbi.nlm.nih.gov/gene/851811">YDR225W</a>, <a href="http://www.ncbi.nlm.nih.gov/gene/852283">YBL003C</a> and <a href="http://www.ncbi.nlm.nih.gov/gene/855701">YNL030W</a>.</p>
<div id="refs" class="references">
<div id="ref-Hotelling:analysis33">
<p>Hotelling, H., 1933. Analysis of a complex of statistical variables into principal components. Journal of Educational Psychology 24, 417–441.</p>
</div>
<div id="ref-Ng:spectral02">
<p>Ng, A.Y., Jordan, M.I., Weiss, Y., n.d. On spectral clustering: Analysis and an algorithm, in:.</p>
</div>
<div id="ref-Rogers:book11">
<p>Rogers, S., Girolami, M., 2011. A first course in machine learning. CRC Press.</p>
</div>
<div id="ref-Shi:normalized00">
<p>Shi, J., Malik, J., 2000. Normalized cuts and image segmentation. IEEE Transactions on Pattern Analysis and Machine Intelligence 22, 888–905.</p>
</div>
<div id="ref-Tipping:probpca99">
<p>Tipping, M.E., Bishop, C.M., 1999. Probabilistic principal component analysis. Journal of the Royal Statistical Society, B 6, 611–622. <a href="https://doi.org/doi:10.1111/1467-9868.00196" class="uri">https://doi.org/doi:10.1111/1467-9868.00196</a></p>
</div>
</div>
</section></section>
    </div>
  </div>

  <script src="reveal.js/lib/js/head.min.js"></script>
  <script src="reveal.js/js/reveal.js"></script>

  <script>

      // Full list of configuration options available at:
      // https://github.com/hakimel/reveal.js#configuration
      Reveal.initialize({
        // Push each slide change to the browser history
        history: true,
        // Transition style
        transition: 'None', // none/fade/slide/convex/concave/zoom

        // Optional reveal.js plugins
        dependencies: [
          { src: 'reveal.js/lib/js/classList.js', condition: function() { return !document.body.classList; } },
          { src: 'reveal.js/plugin/zoom-js/zoom.js', async: true },
          { src: 'reveal.js/plugin/notes/notes.js', async: true }
        ]
      });
    </script>
    </body>
</html>
